### ISSUE TITLE
[0.8.0Beta17] Issues

### ISSUE THREAD

**Issue by Sophist-UK** [meerk40t/meerk40t#1329]
The latest BETA release of 0.8.0 is Beta 17. This issue is for reporting issues with this specific version of MeerK40t.

Please do all further testing on this release. Please do not report any further bugs with previous beta releases. Do give any other types of feedback, questions, concerns or comments.

Personal note: I admire the optimism expressed in the release notes of this beta: "Hopefully last beta" and "Known errors are down to effectively nothing." I am looking forward to confirming all previously reported issues are fixed and that this version does not have any bugs when I try to use it in production.

**Comment by Sophist-UK** [comment]
This comment has been created as a holding place for the development team to use for a list of any issues not yet corrected (if there are any).

**Comment by RuKeBo** [comment]
This is not new and don't think it's a big deal.
If you edit an objects Properties and change the fill color and uncheck "Immediately classify after colour change", the color of the object in the Operation, assuming it is in one, does not change.  The color in the Elements list does change.  If you leave the "classify" check box checked it will classify it and change the color accordingly. If, after you change the color and do not allow it to classify and it doesn't change the color, "Assign Operation" to the object, even it it is the same one, the color will update and change to the correct one.  It just appears that if it doesn't classify, it doesn't refresh the color.

**Comment by RuKeBo** [comment]
Not new again: If you "Collapse all children" in the Operations panel, everything disappears, it does not show the Operations.  I assumed that this collapse would simply not show any of the  objects listed under a give operation, not hide the operation itself.

**Comment by jpirnay** [comment]
> 

Not sure I can reproduce that - if you click on an operation it just closes this single operation, if you click on the root element of operations it will close all. (BTW same goes for elements and group nodes).

**Comment by RuKeBo** [comment]
Correct.  But right-click on "Operations" (or Elements") and select "Collapse children " and everything disappears under Operations or Elements.

**Comment by Sophist-UK** [comment]
Ah - glad it wasn't me that pointed out that "Known errors are down to effectively nothing." was a bit over optimistic.

* [ ] Group nodes do not show Inkscape labels.
* [ ] Node labels are inconsistent. Images do not show the SVG id but paths do.
* [ ] Path node labels - can we please have the Inkscape label before the SVG id if it exists as this is the name the user will primarily know it by. (Perhaps we can also have `Inkscape-Label(id)` in this case.)
* [ ] Previously reported issue with confusing Raster Wizard functionality still exists. Ability to see expected results when Image ops Raster script settings are applied (aka 0.7 Raster Wizard preview) is still missing. Default 500 DPI which (maybe) is equivalent to step 2 is different from 0.7 default of step-1 if you don't use Raster Script and default of 3 for a raster script. 
* [ ] Camera - select a different camera in the dropdown on the button bar and then click the camera icon - correct camera is often not opened in the camera window.
* [ ] Camera Frames per second slider does nothing except when set to 0. Does it make sense to set as low as zero? Frame rate on my i5 Surface Pro is about 1 per 2 secs. Setting is not persistent either.
* [ ] Discord invite no longer works.
* [ ] Path properties - small X under the stroke/fill colour is not great UI. Can  we use a much bigger border for the colour button to indicate this. Mouse over on colour button changes the border (good) but clears the colour (bad).
* [ ] Path properties - Retrieve button takes a long time or starts an infinite loop (didn't want to wait to find out which).
* [ ] Step repeat functionality in Execute Window still missing.
* [ ] Automated begin and end special ops functionality still missing from Execute Window.

Stopping testing at this point.

**Comment by Sophist-UK** [comment]
Previous beta issues entries still have loads of reported issues unticked. It is possible that these have been fixed and no checked off. Perhaps someone can review these and create a list of unfixed issues carried forwards.

**Comment by jpirnay** [comment]
Automated begin and end routines are now device specific and can be managed in the device configuration window

**Comment by jpirnay** [comment]
- [ ] Gridlines do not recognise flipped x- or y-axis fully
- [ ] changes to device dimensions / orientations are only recognised after a restart

**Comment by jpirnay** [comment]
> Camera Frames per second slider does nothing except when set to 0. Does it make sense to set as low as zero? Frame rate on my i5 Surface Pro is about 1 per 2 secs. Setting is not persistent either.

It should provide a new picture every 5 seconds. The tooltip has been extended to indicate this

**Comment by jpirnay** [comment]
> Previous beta issues entries still have loads of reported issues unticked. It is possible that these have been fixed and no checked off. Perhaps someone can review these and create a list of unfixed issues carried forwards.

As far as bugs were concerned they should have been put forward, will validate it over the next couple of days

**Comment by jpirnay** [comment]
> Path properties - Retrieve button takes a long time or starts an infinite loop (didn't want to wait to find out which).
Depending on the shapes complexity this can indeed take some time, as the shapes area will be calculated as well, which uses polygon representation of the shape. All the samples I had were working, so any 'rogue' pattern would be good to have.

**Comment by jpirnay** [comment]
- [ ] Grid not properly centered around device origin
- [ ] Guide not following all combinations of device axis mirroring
- [ ] Moshi-Device-Config incomplete

**Comment by RuKeBo** [comment]
> > Correct.  But right-click on "Operations" (or Elements") and select "Collapse children " and everything disappears under Operations or Elements.
> 
> This has become now a two-stage-collapse. First it will collapse all open (direct) children, if called a second time it will even hide these.
I am not seeing that is is two stages and not sure why it would collapse all the way to showing nothing.  Would there be a use case where you don't want to show anything under "Operations"? If I have 3 operations; Raster, Engrave, and Cut and then put some paths under each of those


> > Correct.  But right-click on "Operations" (or Elements") and select "Collapse children " and everything disappears under Operations or Elements.
> 
> This has become now a two-stage-collapse. First it will collapse all open (direct) children, if called a second time it will even hide these.

I guess we should define what is considered a child and what is consider the parent.  I assumed that the operation under "Operations" is the parent and the element under "Elements" was the parent then anything under those were children.  If Meerk40t considers Operations and Elements themselves the parent then it would probably collapse the actual Operations or Elements themselves since they are children.  I don't agree with this but that is my opinion, I just think you would always want to see the operations or elements, otherwise, it is miss-leading, to me, and looks like there are no Operations or Elements when they are collapsed.

I also do not see how this is two-stage.  I put 3 operations of Engrave, Cut, and Raster in the Operations, put some paths in those operations, manually expand them all then right-click on "Operation" and select "Collapse children" and everything disappears.  It does not just collapse the paths under the Operations.  It collapses everything to nothing.   If it is supposed to be two-stage, I don't know that it is working like that.  I guess it would depend on what is the parent and child as I mentioned above.  If the parent is the operation itself, then it is collapsing that and not showing it when I "Collapse children" once.  I can never use Collapse children twice since there is never a time when I can get more than one child under an operation.  Maybe there is a situation where this can occur but I can't get it to do that in Operations.   However, I can do this in Elements.  I can have an SVG, then a Group under that, and then paths under the group.  I would think this is two children (Group and paths)  but maybe it is 3 (SVG, Group, paths).  It is treating it like "Operations" and "Elements" are the parents and I don't think it should ever collapse the main one (i.e. the SVG element).  Regardless, if I expand the SVG and the group and then right-click on Elements and select "Collapse children", it all disappears, it does not just collapse the path(s) or not just the Group.

It might help to explain how I expected it to work and I think this is consistent with other software and even how Meerk40t works.  If something in the tree has a child and is collapsed, a plus "+" sign appears next to the parent.  The "+" icon indicates it has a child and can be expanded to show it.  If it is expanded it shows the "-" icon to be able to collapse it.  This also works the same as my example above for Elements if there are more children of the children, you still get a plus "+" icon to expand a collapsed item and a minus "-" on an expanded item to collapse it.  If we agree that these icons represent the ability to expand and collapse items, then "Operations", "Elements", and "Regmarks" themselves are not parents should not be collapsible by Meerk40t because they never show a "+" or "-" icon next to them when they have items within them.  If I cannot collapse them manually then why is MeerK40t allowed to when I use Collapse or Expand children?  I expected Meerk40t to only collapse or expand the children that should be expandable or collapsible manually.  And note that I think that is how it should work and you should not be able to expand or collapse "Operations", "Elements", or "Regmarks".  I personally always want to see if there is anything within them.  However, is there is some odd reason that these should be collapsible and expandable, then I think they should have the "+" and "-" icons next to them for consistency.

**Comment by jpirnay** [comment]
> I guess we should define what is considered a child and what is consider the parent. I assumed that the operation under "Operations" is the parent and the element under "Elements" was the parent then anything under those were children. If Meerk40t considers Operations and Elements themselves the parent then it would probably collapse the actual Operations or Elements themselves since they are children. I don't agree with this but that is my opinion, I just think you would always want to see the operations or elements, otherwise, it is miss-leading, to me, and looks like there are no Operations or Elements when they are collapsed.

I get your point and i don't fundamentally disagree, there are some edge cases though, where you might want to have better view of certain parts of the tree and want to reduce the ops /elements even more (eg having quite a lot of groups/root objects under the elements tree)
We could even make it user configurable... But we have a agreed a feature freeze until the next official release, which is not far away imho, so for now this needs to go to the backlog (but noted)

**Comment by jpirnay** [comment]
> Path node labels - can we please have the Inkscape label before the SVG id if it exists as this is the name the user will primarily know it by. (Perhaps we can also have Inkscape-Label(id) in this case.)

The ability to have user-defined node-formatting patterns is in the backlog for the next release after a stable 0.8

**Comment by jpirnay** [comment]
>  Step repeat functionality in Execute Window still missing

True this has been temporarily disabled as the underlying functionality of re-defining homes is currently revised to allow more flexibility.
If you are dependent on this feature then you need either circumvent the issue by just copy / paste the to be repeated elements or continue using v 0.77 until this feature has been reintroduced

**Comment by Sophist-UK** [comment]
> > Path properties - Retrieve button takes a long time or starts an infinite loop (didn't want to wait to find out which).
> Depending on the shapes complexity this can indeed take some time, as the shapes area will be calculated as well, which uses polygon representation of the shape. All the samples I had were working, so any 'rogue' pattern would be good to have.

It was a combined path with a half-dozen tiny circles - so nothing that complicated.

**Comment by Sophist-UK** [comment]
> > Path node labels - can we please have the Inkscape label before the SVG id if it exists as this is the name the user will primarily know it by. (Perhaps we can also have Inkscape-Label(id) in this case.)
> The ability to have user-defined node-formatting patterns is in the backlog for the next release after a stable 0.8

I am **not** asking for user-defined node-formatting patterns - IMO this would be a unnecessary feature where there are much bigger issues to work on. I am simply asking for consistency of fixed formatting patterns - and I consider inconsistency to be a minor bug.

**Comment by Sophist-UK** [comment]
> > I guess we should define what is considered a child and what is consider the parent. I assumed that the operation under "Operations" is the parent and the element under "Elements" was the parent then anything under those were children. If Meerk40t considers Operations and Elements themselves the parent then it would probably collapse the actual Operations or Elements themselves since they are children. I don't agree with this but that is my opinion, I just think you would always want to see the operations or elements, otherwise, it is miss-leading, to me, and looks like there are no Operations or Elements when they are collapsed.
> 
> I get your point and i don't fundamentally disagree, there are some edge cases though, where you might want to have better view of certain parts of the tree and want to reduce the ops /elements even more (eg having quite a lot of groups/root objects under the elements tree) We could even make it user configurable... But we have a agreed a feature freeze until the next official release, which is not far away imho, so for now this needs to go to the backlog (but noted)

**NO** - this is NOT a feature - this is a bug (and one that was fixed in 0.7 if memory serves me correctly). When you close all, the Operations, Elements and Registration Marks nodes should NOT be closed i.e. the next level down should remain visible just like in 0.7. **BUT** if you do decide that the Operations/Elements/Registration nodes should be closed, then they definitely need a +/- symbol next to them.

**Comment by Sophist-UK** [comment]
> > Camera Frames per second slider does nothing except when set to 0. Does it make sense to set as low as zero? Frame rate on my i5 Surface Pro is about 1 per 2 secs. Setting is not persistent either.
> 
> It should provide a new picture every 5 seconds. The tooltip has been extended to indicate this. Persistence works for me.

So what is the purpose of this slider then (as so far I have not been able to decide what its purpose is by experimentation)?

**Comment by jpirnay** [comment]
<img width="113" alt="image" src="https://user-images.githubusercontent.com/2670784/190542398-9a375289-88ce-4f66-94ae-9dd5bd479cf1.png">
Price to pay: about 10 pixels of screen estate

**Comment by jpirnay** [comment]
> > > Camera Frames per second slider does nothing except when set to 0. Does it make sense to set as low as zero? Frame rate on my i5 Surface Pro is about 1 per 2 secs. Setting is not persistent either.
> > 
> > 
> > It should provide a new picture every 5 seconds. The tooltip has been extended to indicate this. Persistence works for me.
> 
> So what is the purpose of this slider then (as so far I have not been able to decide what its purpose is by experimentation)?

It decides the refresh rate of the camera picture, the more you go to the right, the more 'realtime' the picture will become (at the price of CPU consumption) - I personally use the lowest framerate that is still providing 'acceptable' (to me) results, 1 fps, 2 fps max

**Comment by tatarize** [comment]
> Previously reported issue with confusing Raster Wizard functionality still exists. Ability to see expected results when Image ops Raster script settings are applied (aka 0.7 Raster Wizard preview) is still missing. Default 500 DPI which (maybe) is equivalent to step 2 is different from 0.7 default of step-1 if you don't use Raster Script and default of 3 for a raster script.

This is done already it's actually visible in scene. That's what replaced raster-wizard scene, actual scene. The image was modified to apply that stuff during pre-processing since it could be sure to be right otherwise with different devices and different dpi. The default DPI is step 2 (on a 1mil-step device). 0.7 was switched to step 2 too. That was default for most because 1000 dpi was too harsh and burned things more. The raster-wizard I think doesn't mess with the DPI it was inconsistent before with the scripts where some had 3 and others had 2 by default.

**Comment by jpirnay** [comment]
> I am not asking for user-defined node-formatting patterns - IMO this would be a unnecessary feature where there are much bigger issues to work on. I am simply asking for consistency of fixed formatting patterns - and I consider inconsistency to be a minor bug.

I disagree, there are more usecases and personal preferences than just a single one. I e.g. have no use for Svg-ids or labels in my workflow - they take too much space and hide more relevant information for me - I could consider this an annoyance (as I do) or as a 'minor bug' (which it is not). Others will find them very useful, ymmv.

**Comment by Sophist-UK** [comment]
> > I am not asking for user-defined node-formatting patterns - IMO this would be a unnecessary feature where there are much bigger issues to work on. I am simply asking for consistency of fixed formatting patterns - and I consider inconsistency to be a minor bug.
> 
> I disagree, there are more usecases and personal preferences than just a single one. I e.g. have no use for Svg-ids or labels in my workflow - they take too much space and hide more relevant information for me - I could consider this an annoyance (as I do) or as a 'minor bug' (which it is not). Others will find them very useful, ymmv.

So you refuse to accept a reported issue as a bug (when it is removing existing function) because it is not your personal preference?

**Comment by Sophist-UK** [comment]
> > Previously reported issue with confusing Raster Wizard functionality still exists. Ability to see expected results when Image ops Raster script settings are applied (aka 0.7 Raster Wizard preview) is still missing. Default 500 DPI which (maybe) is equivalent to step 2 is different from 0.7 default of step-1 if you don't use Raster Script and default of 3 for a raster script.
> 
> This is done already it's actually visible in scene. That's what replaced raster-wizard scene, actual scene. The image was modified to apply that stuff during pre-processing since it could be sure to be right otherwise with different devices and different dpi. The default DPI is step 2 (on a 1mil-step device). 0.7 was switched to step 2 too. That was default for most because 1000 dpi was too harsh and burned things more. The raster-wizard I think doesn't mess with the DPI it was inconsistent before with the scripts where some had 3 and others had 2 by default.

AFAIK 0.7.7 still has default Raster Wizard step as 3. But this is still as confusing as heck. In image previously saved at step 3 in 0.7.7 still shows up as 500 DPI. And in 0.7 the Wizard shows you a reasonable size image regardless of the scene zoom whereas  on 0.8 you need to zoom the scene in on the image to see the consequences. I also asked for some sort of reset button - you can abort the 0.7 wizard before executing it - there is no way to experiment with settings in 0.8 and then revert them to the way they were before.

**Comment by Sophist-UK** [comment]
> > > > Camera Frames per second slider does nothing except when set to 0. Does it make sense to set as low as zero? Frame rate on my i5 Surface Pro is about 1 per 2 secs. Setting is not persistent either.
> > > 
> > > 
> > > It should provide a new picture every 5 seconds. The tooltip has been extended to indicate this. Persistence works for me.
> > 
> > 
> > So what is the purpose of this slider then (as so far I have not been able to decide what its purpose is by experimentation)?
> 
> It decides the refresh rate of the camera picture, the more you go to the right, the more 'realtime' the picture will become (at the price of CPU consumption) - I personally use the lowest framerate that is still providing 'acceptable' (to me) results, 1 fps, 2 fps max

Framerate does not change when you move the slider (except when on 0 when the picture freezes) - I get 1 frame every 5 seconds as stated above, and the framerate doesn't change. That is the bug report.

**Comment by jpirnay** [comment]
> 

No I don't deny your request to have a different display, the solution you are proposing is not the one I see as there are different possibly conflicting demands that require a different approach..

**Comment by jpirnay** [comment]
> Framerate does not change when you move the slider (except when on 0 when the picture freezes) - I get 1 frame every 5 seconds as stated above, and the framerate doesn't change. That is the bug report.

Thank you, confirmed due to an old setting in place that may override this behaviour if in the cfg file. If you want to see its effect look for ``fps=1`` in the config file. Replace it to something like fps=40.

**Comment by tatarize** [comment]
* The step rate in 0.7.7 for raster wizard actually depends on the script applied.

* Type `set -p camera/0 fps 60` to undo the current setting. It got split from the other one but is still stuck low for the scene. The slider controls the camera fps not the scene fps.

**Comment by Sophist-UK** [comment]
> No I don't deny your request to have a different display, the solution you are proposing is not the one I see as there are different possibly conflicting demands that require a different approach..

No - you are refusing to accept it as a bug (which will be fixed in 0.8) and treating it as a feature request for 0.9.

**Comment by tatarize** [comment]
> Ah - glad it wasn't me that pointed out that "Known errors are down to effectively nothing." was a bit over optimistic.

Known serious errors are, there are some cosmetic issues but actual kick you in the teeth with not working *are* actually reduced pretty heavily. Some cosmetic issues that don't produce crashes will probably be swept under the rug. Though I'll likely need to go through them all and actually decide to dismiss them, before a full release is issued. There's still some scut housekeeping like updating the translation templates etc. Rereviewing the crash logs and some other stuff.

Known errors are down to nothing. Unknown errors are of an unknown quantity.

**Comment by tatarize** [comment]
The fixing of the failure to start from the current location identified another error where the HomeCut was being optimized. When it was close to 0,0 it would jot back for a home and then continue with the laser. This from having `Home Before` set. Both should be fixed in Beta18.

**Comment by Sophist-UK** [comment]
> Known serious errors are, there are some cosmetic issues but actual kick you in the teeth with not working _are_ actually reduced pretty heavily. Some cosmetic issues that don't produce crashes will probably be swept under the rug. Though I'll likely need to go through them all and actually decide to dismiss them, before a full release is issued. There's still some scut housekeeping like updating the translation templates etc. Rereviewing the crash logs and some other stuff.
> 
> Known errors are down to nothing. Unknown errors are of an unknown quantity.

Cosmetic issues ARE important and IMO should **not** "be swept under the rug".

As I have said elsewhere, 0.8 is getting to be close to production ready - but IMO the confusing UI for the new image rasterisation functionality is so confusing as to be a significant usability issue.

(But I accept that Step Repeat can wait until 0.8.1.)

**Comment by tatarize** [comment]
> AFAIK 0.7.7 still has default Raster Wizard step as 3. But this is still as confusing as heck. In image previously saved at step 3 in 0.7.7 still shows up as 500 DPI. And in 0.7 the Wizard shows you a reasonable size image regardless of the scene zoom whereas on 0.8 you need to zoom the scene in on the image to see the consequences. I also asked for some sort of reset button - you can abort the 0.7 wizard before executing it - there is no way to experiment with settings in 0.8 and then revert them to the way they were before.

The default depended on the script you use. It would give you whatever the script said to use. In 0.8.x it does not set a default. It actually just uses the DPI you give it. So it doesn't have a clear requirement here.

A bit of background. The RasterWizard would actualize the image in a concrete fashion. So the image in the scene was actually that the actual end result pixels of the actual thing you would raster. This required that X steps equaled 1 pixel. So a step of 2 would be sized such that 2 mils would equal 1 pixel. Then it would lock that in place. In 0.8.x because we fully support other drivers for the laser, we do not actually perform the action that rasterwizard says. We just do it in the scene as a soft action. The image nodes *always* have access to the original image and each time you mess with something it applies the modified script to the original image and gives you the result. This is a simulated result you see in the scene and it's directly related to the view of the image you would see in RasterWizard 0.7.x/0.6.x it actually applies this real script during the cut planning. When you send this to the laser rather than just do "actualize" if it needed to be resized at that point it performs the rasterwizard script inside the cutplanning process. This is because only during cutplanning do we know exactly what backend we are planning for so we know the actual sizes things need to be.

This is also why we prefer DPI to steps. Steps implies discrete elements of distance like steps of a stepper motor, but we don't actually know how big that is. We lost the 1 mil = 1 step assumption. So 2 steps on a galvo laser may actually be basically any size (depends on the lens), but 2 steps on a K40 is about 2 mils (belt stretching aside). In order to accommodate different lasers we needed units that were directly references to physical units. Dots-Per-Inch relates to Inches which have a concrete size (exactly 2.54 cm). We cannot actually use steps on a galvo laser though. During testing Sandor had to set Steps to about 20 to get something the size of a postage stamp. The reason is that given the units there a step was about 0.02 mils. However, a DPI of 500 for a similarly sized image gives us the same quality regardless what the native units are for the device.

I am quite happy to discuss how to make the GUI more user-friendly and I get that some of the required changes are confusing. But, the reasons for doing this are numerous and quite required. The old version 0.7.x could not actually account for stretched belts. Since 1 step = 1 mil was hard-baked into things. But, if you want an image that is correctly sized regardless that your belts are 10% stretched and your 1 step goes 1.1 mils, it *needed* to be coded this way. 

---

I did the accounting of previous versions of these errata and checked for the ones that were completed and asked. See #1332 to make sure nothing falls through the cracks. A couple of them I fixed while doing the list like the `control+mousewheel zoom`.

**Comment by tatarize** [comment]
> So you refuse to accept a reported issue as a bug (when it is removing existing function) because it is not your personal preference?

Point of fact, that's actually just a defect if anything at all. It's not broken it works the way it's designed but it may be designed sub-optimally which is defect not bug. But, he wrote a whole you can go ahead and make those labels say anything you want thing so it's likely fine in the long run. I sort of considered this a bit feature creep since it's moving stuff that is generally dynamic into letting people theme their nodes.

This was done because Galvo lasers actually use kHz and have frequencies. This is a fundamental part of the operation for Galvo lasers but would be utterly pointless to a K40 user, it could even be confusing to them. So this was generally changed so in Galvo/Balor you were given frequency but in M2 Nano you were not.

**Comment by Sophist-UK** [comment]
If that is your definition of a bug, it stinks. If a user is reporting an issue with a 0.8 beta version cf. functionality of a 0.7 version, it is from a user perspective a bug (regardless of the developer intent) and unless there is a good reason for developers not to treat it as a bug then they should do so.

I have no issues with labels showing device-specific information, but that does not invalidate the issue / requirement as reported by me as a user - labels still need to be consistent and show information generally useful to a reasonable proportion of the user base. (And since I am literally doing more testing and reporting more issues than all other users put together, I would hope that my needs carry some extra weight. Or are you going to drive me away from doing beta testing in a similar way to how you drove me away as a developer?)

**Comment by Sophist-UK** [comment]
# Crash in Execute Window on clear of Merge Passes checkbox:
```
MeerK40t crash log. Version: 0.8.0025 Beta17 on Windows: Python 3.7.9: AMD64 - wxPython: 4.1.1 msw (phoenix) wxWidgets 3.1.5
Traceback (most recent call last):
  File "meerk40t\kernel\kernel.py", line 1742, in schedule_run
  File "meerk40t\kernel\kernel.py", line 1957, in process_queue
  File "meerk40t\kernel\kernel.py", line 1927, in _process_signal_queue
  File "meerk40t\gui\choicepropertypanel.py", line 845, in listen_to_myself
RuntimeError: wrapped C/C++ object of type CheckBox has been deleted
```

**Comment by tatarize** [comment]
Hm. Mine isn't crashing there. The bug says that a signal is issued with the listen_to_myself listener still attached. But, if execute window is opened it wouldn't. Let me see. Maybe shut it down and reopen it. -- yeah, that did it.

The lifecycle is there. It won't do it the first time. You need to open, close, and reopen Execute Window then change the state of Merge Passes.

**Comment by tatarize** [comment]
Great catch. It's a little harder to invoke than just uncheck a box though.

It would actually plant that bomb any time. So if you loaded Execute Job, then dechecked merge passes in LaserPane it should also crash there.

**Comment by Sophist-UK** [comment]
Issues when attempting to use 0.8.0B17 for production use.
* [ ] Execute / Simulation of selected operations does not include default special ops.
* [ ] Simulation - scene not updated when you move the playback slider - but does update if you do e.g. a zoom scene - so an update signal may be missing. (May be a consequence of previous crash - will restart and retest.)
* [ ] Simulation - playback speed of e.g. 5000% does not seem to go faster.
* [ ] Execute - Copy button stays as Copy after copy rather than becoming Blob. Ops get duplicated. (May be a consequence of previous crash.)
* [ ] Execute - Automated finish beep occurs before burn has actually finished. Interrupt popup happens before previous ops have actually finished burning.
* [ ] Close MK after burn has completed gives "The device is actively sending data. Really quit?".
* [ ] Red background behind scene when burn is happening is missing.

**Comment by Sophist-UK** [comment]
* [ ] Each burn is now shown as a single item in spooler even with special ops and no merge ops. I can see this is a good thing when e.g. you are running several lasers in parallel and you want to see each job as a single line. However this means that you cannot e.g. set a job upon with several passes on engrave and then several passes on cut, and kill off individual passes once you have enough burn. This was a **VERY** useful ability in 0.7 and I miss it already in 0.8. Can we please have the ability to select a burn in the spooler and open a window showing individual "operations" bearing in mind merges etc. and e.g. cancel later ops?
* [ ] **Execute - multiple operation passes not burned**
* [ ] Spooler - Estimate column shows elapsed time not estimated total time.
* [ ] Spooler - Name column should show project name or similar (for multiple projects to multiple lasers simultaneously) - number of items should be a separate column.
* [ ] Spooler - Device column needed (for multiple simultaneous lasers).
* [ ]

**Comment by Sophist-UK** [comment]
> The crash kills the scheduler so anything that gets scheduled is somewhat likely to not work, also a lot of console commands break after a crash. Crashes break a the kernel basically which doesn't easily recover. The Bee

Is it possible for the crash handler to reset the kernel so that things all work - or will this e.g. break burns in progress? How about fixing things that do not effect a burn in progress?

**Comment by Sophist-UK** [comment]
No more testing from me now until all essential issues above have been fixed in beta 18.

**Comment by Sophist-UK** [comment]
Closed because beta18 is available.
